src/libs/print_settings.c

```diff
@@ -1388,46 +1389,31 @@ static gboolean _new_printer_callback(gpointer user_data)
 {
   static int count = 0;
   dt_printer_discovered_t *printer = (dt_printer_discovered_t *)user_data;
-  const dt_lib_module_t *self = (dt_lib_module_t *)printer->user_data;
+  dt_lib_module_t *self = (dt_lib_module_t *)printer->user_data;
   const dt_lib_print_settings_t *d = (dt_lib_print_settings_t *)self->data;
 
-  char *default_printer = dt_conf_get_string("plugins/print/print/printer");
+  const char *const default_printer =
+    dt_conf_get_string_const("plugins/print/print/printer");
 
   g_signal_handlers_block_by_func(G_OBJECT(d->printers),
-                                  G_CALLBACK(_printer_changed), NULL);
+                                  G_CALLBACK(_printer_changed), self);
```

Comment:
This is indeed always finicky which is why for bauhaus widgets you can use
```
      ++darktable.gui->reset;
      dt_bauhaus_combobox_set etc
      --darktable.gui->reset;
````
to block all value-changed events.

Reply:
Ah, that makes a lot of sense. Thank you as always.

Reply:
pushed one more commit to tidy this!